### PR TITLE
Remove unneeded null type from properties.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
```
1) Cake\Test\TestCase\Mailer\Transport\SmtpTransportTest::testSerializeCleanupSocket
Cake\Network\Socket::isConnected(): bool was not expected to be called more than once.

/home/runner/work/cakephp/cakephp/src/Mailer/Transport/SmtpTransport.php:117
/home/runner/work/cakephp/cakephp/tests/TestCase/Mailer/Transport/SmtpTransportTest.php:867
```

I don't understand this failure. I put a debug in `SmtpTransport::__wakeup()` and it doesn't show up when `unserialize()` is called.

File: tests/TestCase/Mailer/Transport/SmtpTransportTest.php [hunk 1]
```diff
@@ -855,16 +856,18 @@ public function testSerializeCleanupSocket(): void
...
             ->with("EHLO localhost\r\n");
+        $this->socket->expects($this->once())
+            ->method('isConnected')
+            ->willReturn(true);
 
         $smtpTransport = new SmtpTestTransport();
         $smtpTransport->setSocket($this->socket);
         $smtpTransport->connect();
 
+        $this->assertTrue($smtpTransport->connected());
```
Comment (othercorey) [inline tests/TestCase/Mailer/Transport/SmtpTransportTest.php]: The `connect()` call above already calls `connected()` so the total calls to isConnected() should be 2.

Comment (ADmad) [reply tests/TestCase/Mailer/Transport/SmtpTransportTest.php]: Ah, silly me

Comment (ADmad) [reply tests/TestCase/Mailer/Transport/SmtpTransportTest.php]: I don't think I can test what I want to using the mock anyway. Can't expect SmtpTransport instance created after unserialization to have the same mock instance of the socket.

**Pull request merged**